aten/src/ATen/native/SpectralOps.cpp

```diff
@@ -257,19 +281,35 @@ Tensor stft(const Tensor& self, const int64_t n_fft, const optional<int64_t> hop
   if (window_.defined()) {
     input = input.mul(window_);
   }
-  // rfft and transpose to get (batch x fft_size x num_frames)
-  auto out = input.rfft(1, normalized, onesided).transpose_(1, 2);
+
+  // FFT and transpose to get (batch x fft_size x num_frames)
+  const bool complex_fft = input.is_complex();
+  const auto onesided = onesidedOpt.value_or(!complex_fft);
+
+  Tensor out;
+  if (complex_fft) {
+    TORCH_CHECK(!onesided, "Cannot have onesided output if window or input is complex");
+    out = at::native::fft(at::view_as_real(input), 1, normalized);
+  } else {
+    out = at::native::rfft(input, 1, normalized, onesided);
+  }
+  out.transpose_(1, 2);
+
   if (self.dim() == 1) {
-    return out.squeeze_(0);
+    out.squeeze_(0);
+  }
+
+  if (return_complex) {
+    return at::view_as_complex(out);
   } else {
     return out;
   }
 }
 
 Tensor istft(const Tensor& self, const int64_t n_fft, const optional<int64_t> hop_lengthOpt,
```

Comment:
Comment here, like in the above

Reply:
For this function I'm also curious about preserving the current behavior and supporting new complex behavior. How does SciPy's istft handle the case where onesided is false for float and complex inputs?

Reply:
For SciPy, if `input_onesided` is `False` it does a full `ifft` and always returns complex input.